presto-hive/src/main/java/io/prestosql/plugin/hive/ViewReaderUtil.java

```diff
@@ -0,0 +1,229 @@
+    private static Logger log = Logger.get(io.prestosql.plugin.hive.ViewReaderUtil.class);
+
+    public static boolean isPrestoView(Table table)
+    {
+        return "true".equals(table.getParameters().get(PRESTO_VIEW_FLAG));
+    }
+
+    public static boolean isHiveOrPrestoView(Table table)
+    {
+        return table.getTableType().equals(TableType.VIRTUAL_VIEW.name());
+    }
+
+    public static boolean canDecodeView(Table table)
+    {
+        // we can decode Hive or Presto view
+        return table.getTableType().equals(VIRTUAL_VIEW.name());
+    }
+
+    public static String encodeViewData(ConnectorViewDefinition definition)
+    {
+        byte[] bytes = VIEW_CODEC.toJsonBytes(definition);
+        String data = Base64.getEncoder().encodeToString(bytes);
+        return VIEW_PREFIX + data + VIEW_SUFFIX;
+    }
+
+    /**
+     * Supports decoding of Presto views
+     */
+    public static class PrestoViewDecoder
+            implements ViewReader
+    {
+        @Override
+        public ConnectorViewDefinition decodeViewData(String viewData, Table table, CatalogName catalogName)
+        {
+            checkCondition(viewData.startsWith(VIEW_PREFIX), HIVE_INVALID_VIEW_DATA, "View data missing prefix: %s", viewData);
+            checkCondition(viewData.endsWith(VIEW_SUFFIX), HIVE_INVALID_VIEW_DATA, "View data missing suffix: %s", viewData);
+            viewData = viewData.substring(VIEW_PREFIX.length());
+            viewData = viewData.substring(0, viewData.length() - VIEW_SUFFIX.length());
+            byte[] bytes = Base64.getDecoder().decode(viewData);
+            return VIEW_CODEC.fromJson(bytes);
+        }
+    }
+
+    /**
+     * Class to decode Hive view definitions
+     */
+    public static class HiveViewDecoder
+            implements ViewReader
+    {
+        private final HiveMetastoreClient mscClient;
+        private final TypeManager typeManager;
+
+        public HiveViewDecoder(HiveMetastoreClient mscClient, TypeManager typemanager)
+        {
+            this.mscClient = mscClient;
+            this.typeManager = requireNonNull(typemanager, "metadata is null");
+        }
+
+        @Override
+        public ConnectorViewDefinition decodeViewData(String viewSql, Table table, CatalogName catalogName)
+        {
+            try {
+                HiveToRelConverter hiveToRelConverter = HiveToRelConverter.create(mscClient);
+                RelNode rel = hiveToRelConverter.convertView(table.getDatabaseName(), table.getTableName());
+                RelToPrestoConverter rel2Presto = new RelToPrestoConverter();
+                String prestoSql = rel2Presto.convert(rel);
+                RelDataType rowType = rel.getRowType();
+                List<ConnectorViewDefinition.ViewColumn> columns = new ArrayList<>(rowType.getFieldCount());
+                for (RelDataTypeField field : rowType.getFieldList()) {
+                    log.debug("Adding column %s, %s, %s, %s", field.getName(),
```

Comment:
Is this useful to leave in the code? If so, we should expand it to include the view name. Otherwise, if it's just for development, please remove it.